src/NuGet.Core/NuGet.Build.Tasks/BuildTasksUtility.cs

```diff
@@ -97,5 +100,80 @@ public static void AddPropertyIfExists(IDictionary<string, string> properties, s
             ProjectStyle.Standalone,
             ProjectStyle.ProjectJson
         };
+
+        /// <summary>
+        /// Determines the restore style of a project.
+        /// </summary>
+        /// <param name="restoreProjectStyle">An optional user supplied restore style.</param>
+        /// <param name="hasPackageReferenceItems">A <see cref="bool"/> indicating whether or not the project has any PackageReference items.</param>
+        /// <param name="projectJsonPath">An optional path to the project's project.json file.</param>
+        /// <param name="projectDirectory">The full path to the project directory.</param>
+        /// <param name="projectName">The name of the project file.</param>
+        /// <param name="log">An <see cref="NuGet.Common.ILogger"/> object used to log messages.</param>
+        /// <returns>A <see cref="Tuple{ProjectStyle, Boolean}"/> containing the project style and a value indicating if the project is using a style that is compatible with PackageReference.
+        /// If the value of <paramref name="restoreProjectStyle"/> is not empty and could not be parsed, <code>null</code> is returned.</returns>
+        public static (ProjectStyle ProjectStyle, bool IsPackageReferenceCompatibleProjectStyle) GetProjectRestoreStyle(string restoreProjectStyle, bool hasPackageReferenceItems, string projectJsonPath, string projectDirectory, string projectName, Common.ILogger log)
+        {
+            ProjectStyle projectStyle;
+
+            // Allow a user to override by setting RestoreProjectStyle in the project.
+            if (!string.IsNullOrWhiteSpace(restoreProjectStyle))
+            {
+                if (!Enum.TryParse(restoreProjectStyle, ignoreCase: true, out projectStyle))
+                {
+                    // Any value that is not recognized is treated as Unknown
+                    projectStyle = ProjectStyle.Unknown;
+                }
+            }
+            else if (hasPackageReferenceItems)
+            {
+                // If any PackageReferences exist treat it as PackageReference. This has priority over project.json.
+                projectStyle = ProjectStyle.PackageReference;
+            }
+            else if (!string.IsNullOrWhiteSpace(projectJsonPath))
+            {
+                // If this is not a PackageReference project check if project.json or projectName.project.json exists.
+                projectStyle = ProjectStyle.ProjectJson;
+            }
+            else if (ProjectHasPackagesConfigFile(projectDirectory, projectName))
```

Comment:
Do we need a null check at least on `projectDirectory`?  In the original task, an undefined value would not throw an exception.  [Here](https://github.com/NuGet/NuGet.Client/pull/3131/files#diff-dc13cfd7bd242d1b20b434573a3f344aR162) it would.

Reply:
Technically if this is called by the task, the value would never be null.  But later on when I call it in my other code I could accidentally call it incorrectly.  I might as well add a null check in `ProjectHasPackagesConfigFile()`.

On a related note, should I make this method `internal`?  Everything else in in this class is `public` but I don't see why we should make this public, thoughts?